src/test/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplateTests.java

```diff
@@ -711,6 +698,100 @@ void shouldReturnSortFields() {
 				.verifyComplete();
 	}
 
+	@Test // DATAES-623
+	public void shouldReturnObjectsForGivenIdsUsingMultiGet() {
+		SampleEntity entity1 = randomEntity("test message 1");
+		entity1.rate = 1;
+		index(entity1);
+		SampleEntity entity2 = randomEntity("test message 2");
+		entity2.rate = 2;
+		index(entity2);
+
+		NativeSearchQuery query = new NativeSearchQueryBuilder() //
+				.withIds(Arrays.asList(entity1.getId(), entity2.getId())) //
+				.build();
+
+		template.multiGet(query, SampleEntity.class, IndexCoordinates.of(DEFAULT_INDEX)) //
+				.as(StepVerifier::create) //
+				.expectNext(entity1, entity2) //
+				.verifyComplete();
+	}
+
+	@Test // DATAES-623
+	public void shouldReturnObjectsForGivenIdsUsingMultiGetWithFields() {
+		SampleEntity entity1 = randomEntity("test message 1");
+		entity1.rate = 1;
+		index(entity1);
+		SampleEntity entity2 = randomEntity("test message 2");
+		entity2.rate = 2;
+		index(entity2);
+
+		NativeSearchQuery query = new NativeSearchQueryBuilder() //
+				.withIds(Arrays.asList(entity1.getId(), entity2.getId())) //
+				.withFields("message") //
+				.build();
+
+		template.multiGet(query, SampleEntity.class, IndexCoordinates.of(DEFAULT_INDEX)) //
+				.as(StepVerifier::create) //
+				.expectNextCount(2) //
+				.verifyComplete();
+	}
+
+	@Test // DATAES-623
+	public void shouldDoBulkUpdate() {
+		SampleEntity entity1 = randomEntity("test message 1");
+		entity1.rate = 1;
+		index(entity1);
+		SampleEntity entity2 = randomEntity("test message 2");
+		entity2.rate = 2;
+		index(entity2);
+
+		IndexRequest indexRequest1 = new IndexRequest();
+		indexRequest1.source("message", "updated 1");
+		UpdateQuery updateQuery1 = new UpdateQueryBuilder() //
+				.withId(entity1.getId()) //
+				.withIndexRequest(indexRequest1).build();
+
+		IndexRequest indexRequest2 = new IndexRequest();
+		indexRequest2.source("message", "updated 2");
+		UpdateQuery updateQuery2 = new UpdateQueryBuilder() //
+				.withId(entity2.getId()) //
+				.withIndexRequest(indexRequest2).build();
+
+		List<UpdateQuery> queries = Arrays.asList(updateQuery1, updateQuery2);
+		template.bulkUpdate(queries, IndexCoordinates.of(DEFAULT_INDEX)).block();
+
+		NativeSearchQuery getQuery = new NativeSearchQueryBuilder() //
+				.withIds(Arrays.asList(entity1.getId(), entity2.getId())) //
+				.build();
+		template.multiGet(getQuery, SampleEntity.class, IndexCoordinates.of(DEFAULT_INDEX)) //
+				.as(StepVerifier::create) //
+				.expectNextMatches(entity -> entity.getMessage().equals("updated 1")) //
+				.expectNextMatches(entity -> entity.getMessage().equals("updated 2")) //
+				.verifyComplete();
```

Comment:
is this order guaranteed?

Reply:
Quote from [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-multi-get.html#multi-get-api-response-body):
``
The response includes a docs array that contains the documents in the order specified in the request. 
``